x-pack/plugins/cloud_security_posture/public/pages/findings/latest_findings_by_resource/resource_findings/use_resource_findings.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useQuery } from 'react-query';
+import { lastValueFrom } from 'rxjs';
+import { IEsSearchResponse } from '@kbn/data-plugin/common';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { useKibana } from '../../../../common/hooks/use_kibana';
+import { showErrorToast } from '../../latest_findings/use_latest_findings';
+import type { CspFinding, FindingsBaseEsQuery, FindingsQueryResult } from '../../types';
+
+interface UseResourceFindingsOptions extends FindingsBaseEsQuery {
+  resourceId: string;
+}
+
+export type ResourceFindingsResult = FindingsQueryResult<
+  ReturnType<typeof useResourceFindings>['data'] | undefined,
+  unknown
+>;
+
+export const getResourceFindingsQuery = ({
+  index,
+  query,
+  resourceId,
+}: UseResourceFindingsOptions): estypes.SearchRequest => ({
+  index,
+  body: {
+    query: {
+      ...query,
+      bool: {
+        ...query?.bool,
+        filter: [...(query?.bool?.filter || []), { term: { 'resource_id.keyword': resourceId } }],
+      },
+    },
+  },
```

Comment:
this is where we say this query is limited to a `resourceId`

might change this later to just be `merge(a,b)` to read nicer